test/metrics/metrics.test.ts

```diff
@@ -146,7 +154,7 @@ test.serial('Test http request duration metrics', async (t) => {
 test.serial('Test data provider requests metrics', async (t) => {
   const metricsMap = await t.context.testAdapter.getMetrics()
   const expectedLabel = `{provider_status_code="200",method="GET",app_name="TEST",app_version="${version}"}`
-  t.is(metricsMap.get(`data_provider_requests${expectedLabel}`), 1)
+  t.is(metricsMap.get(`data_provider_requests${expectedLabel}`), 9)
```

Comment:
I may have missed something but what's causing these metrics to have a jump in their numbers?

Reply:
Because this test didn't use to use a mocked system clock, and now that it does in the methods we use in the testing classes to wait for results we also account for the background executes. That makes it so each time we gradually advance the clock by iterating over the scheduled async calls, we could be firing more background executes in the meantime

Reply:
Oh ok I see. So we are actually sending more than just the one request because of the background execute.